packages/shared/src/hooks/logging-hook.ts

```diff
@@ -0,0 +1,70 @@
+import type { OpenFeatureError } from '../errors';
+import type { BaseHook } from './hook';
+import type { BeforeHookContext, HookContext, HookHints } from './hooks';
+import type { FlagValue, EvaluationDetails } from '../evaluation';
+
+import { DefaultLogger, SafeLogger } from '../logger';
+
+type LoggerPayload = Record<string, unknown>;
+
+const DOMAIN_KEY = 'domain';
+const PROVIDER_NAME_KEY = 'provider_name';
+const FLAG_KEY_KEY = 'flag_key';
+const DEFAULT_VALUE_KEY = 'default_value';
+const EVALUATION_CONTEXT_KEY = 'evaluation_context';
+const ERROR_CODE_KEY = 'error_code';
+const ERROR_MESSAGE_KEY = 'error_message';
+const REASON_KEY = 'reason';
+const VARIANT_KEY = 'variant';
+const VALUE_KEY = 'value';
+
+export class LoggingHook implements BaseHook {
+  readonly includeEvaluationContext: boolean = false;
+  readonly logger = new SafeLogger(new DefaultLogger(true, true));
+
+  constructor(includeEvaluationContext: boolean = false) {
+    this.includeEvaluationContext = !!includeEvaluationContext;
+  }
+
+  before(hookContext: BeforeHookContext): void {
+    const payload: LoggerPayload = { stage: 'before' };
+    this.addCommonProps(payload, hookContext);
+    this.logger.debug(payload);
+  }
+
+  after(hookContext: Readonly<HookContext<FlagValue>>, evaluationDetails: EvaluationDetails<FlagValue>): void {
+    const payload: LoggerPayload = { stage: 'after' };
+
+    payload[REASON_KEY] = evaluationDetails.reason;
+    payload[VARIANT_KEY] = evaluationDetails.variant;
+    payload[VALUE_KEY] = evaluationDetails.value;
+
+    this.addCommonProps(payload, hookContext);
+    this.logger.debug(payload);
+  }
+
+  error(hookContext: Readonly<HookContext<FlagValue>>, error: OpenFeatureError): void {
+    const payload: LoggerPayload = { stage: 'error' };
+
+    payload[ERROR_MESSAGE_KEY] = error.message;
+    payload[ERROR_CODE_KEY] = error.code;
+
+    this.addCommonProps(payload, hookContext);
+    this.logger.error(payload);
+  }
+
+  finally(hookContext: Readonly<HookContext<FlagValue>>, hookHints?: HookHints): void {
```

Comment:
Please make sure to rebase your PR. This method signature has changed recently.

https://github.com/open-feature/js-sdk/pull/1087

Reply:
@beeme1mr This has been addressed with latest commit.